plugins/inputs/http/http.go

```diff
@@ -184,6 +185,23 @@ func (h *HTTP) gatherURL(
 	return nil
 }
 
+func (h *HTTP) setRequestAuth(request *http.Request) error {
+	username, err := h.Username.Get()
+	if err != nil {
+		return fmt.Errorf("getting username failed: %v", err)
+	}
+	defer config.ReleaseSecret(username)
```

Comment:
The (clear-text) secret given to you is a byte-array. To protect this clear-text to be e.g. swapped-out to disk, we protect the memory region and want to release this protection. Furthermore, we want to safely wipe the secret from memory once it is not used anymore. Therefore, we need some indication when to do this (i.e. the release). We could of course also clean this up on garbage-collection, but this would make the plain-text secret to stay in memory for an unforeseeable time...